src/cdk-experimental/ui-patterns/toolbar/toolbar.ts

```diff
@@ -7,237 +7,236 @@
-  nextKey = computed(() => {
+  private readonly _nextKey = computed(() => {
     if (this.inputs.orientation() === 'vertical') {
       return 'ArrowDown';
     }
     return this.inputs.textDirection() === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
   });
 
-  /** The alternate key used to navigate to the previous widget */
-  altPrevKey = computed(() => {
+  /** The alternate key used to navigate to the previous widget. */
+  private readonly _altPrevKey = computed(() => {
     if (this.inputs.orientation() === 'vertical') {
       return this.inputs.textDirection() === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
     }
     return 'ArrowUp';
   });
 
   /** The alternate key used to navigate to the next widget. */
-  altNextKey = computed(() => {
+  private readonly _altNextKey = computed(() => {
     if (this.inputs.orientation() === 'vertical') {
       return this.inputs.textDirection() === 'rtl' ? 'ArrowLeft' : 'ArrowRight';
     }
     return 'ArrowDown';
   });
 
   /** The keydown event manager for the toolbar. */
-  keydown = computed(() => {
+  private readonly _keydown = computed(() => {
     const manager = new KeyboardEventManager();
 
-    const activeItem = this.inputs.activeItem();
-    const isRadioButton = activeItem instanceof RadioButtonPattern;
+    return manager
+      .on(this._nextKey, () => this._next())
+      .on(this._prevKey, () => this._prev())
+      .on(this._altNextKey, () => this._groupNext())
+      .on(this._altPrevKey, () => this._groupPrev())
+      .on(' ', () => this._trigger())
+      .on('Enter', () => this._trigger())
+      .on('Home', () => this._first())
+      .on('End', () => this._last());
+  });
 
-    if (isRadioButton) {
-      manager
-        .on(' ', () => this.selectRadioButton())
-        .on('Enter', () => this.selectRadioButton())
-        .on(this.altNextKey, () => activeItem?.group()?.listBehavior.next())
-        .on(this.altPrevKey, () => activeItem?.group()?.listBehavior.prev());
-    } else {
-      manager.on(this.altNextKey, () => this.listBehavior.next());
-      manager.on(this.altPrevKey, () => this.listBehavior.prev());
+  /** The pointerdown event manager for the toolbar. */
+  private readonly _pointerdown = computed(() => new PointerEventManager().on(e => this._goto(e)));
+
+  /** Navigates to the next widget in the toolbar. */
+  private _next() {
+    const item = this.inputs.activeItem();
+    if (item instanceof ToolbarWidgetGroupPattern) {
+      const {leaveGroup} = item.actions().next(false) ?? {};
+      if (!leaveGroup) return;
     }
 
-    return manager
-      .on(this.prevKey, () => this.listBehavior.prev())
-      .on(this.nextKey, () => this.listBehavior.next())
-      .on('Home', () => this.listBehavior.first())
-      .on('End', () => this.listBehavior.last());
-  });
+    this.listBehavior.next();
+    const newItem = this.inputs.activeItem();
+    if (newItem instanceof ToolbarWidgetGroupPattern) {
+      newItem.actions().first();
+    }
+  }
```

Comment:
Would it make more sense to expose something like "isOnLastItem()"  through the "actions" so the parent can just check that directly instead of having the information come through a return value?